### PR TITLE
[dlc] add resource prepare support

### PR DESCRIPTION
**Pull request by yjjinjie**
Description
add resource_config support
ecsSpec: "{'resource_id':'rg19d2oleg252kke','cpu':2,'memory':8,'gpu':0,'gputype':'','shared_memory':''}"

Checklist
verified on Alibaba PAI DLC

How to test
open alibaba pai DSW or use ECS ,follow the docs to create a file for testing.

### CODE REVIEW

File: examples/trials/mnist-pytorch/config_dlc_resource.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  classArgs:
+    optimize_mode: maximize
+# ref: https://help.aliyun.com/document_detail/203290.html?spm=a2c4g.11186623.6.727.6f9b5db6bzJh4x
+trainingService:
+  platform: dlc
+  type: Worker
+  image: registry-vpc.cn-beijing.aliyuncs.com/pai-dlc/pytorch-training:1.6.0-gpu-py37-cu101-ubuntu18.04
+  jobType: PyTorchJob                             # choices: [TFJob, PyTorchJob]
+  podCount: 1
+  ecsSpec: "{'resource_id':'rg19d2oleg252kke','cpu':2,'memory':8,'gpu':0,'gputype':'','shared_memory':''}"
```
Comment (weidankong) [inline examples/trials/mnist-pytorch/config_dlc_resource.yml:18]: please add some comments on the parameters

File: ts/nni_manager/config/dlc/dlcUtil.py [hunk 2]
```diff
@@ -71,30 +72,65 @@
...
+        else:
+            # job spec
+            spec = JobSpec(
+                type=args.type,
+                image=args.image,
+                pod_count=args.pod_count,
+                ecs_spec=args.ecs_spec,
+            )
+
+
```
Comment (weidankong) [inline ts/nni_manager/config/dlc/dlcUtil.py:124]: too many empty lines.

File: examples/trials/mnist-pytorch/config_dlc_resource.yml (hunk 1 above)
Comment (yjjinjie) [reply examples/trials/mnist-pytorch/config_dlc_resource.yml]: OK

File: ts/nni_manager/config/dlc/dlcUtil.py (hunk 2 above)
Comment (yjjinjie) [reply ts/nni_manager/config/dlc/dlcUtil.py]: OK

**Review by weidankong — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**